tools/istio-iptables/pkg/cmd/root.go

```diff
@@ -15,30 +15,49 @@
 package cmd
 
 import (
+	"fmt"
+	"net"
 	"os"
+	"os/user"
 	"strings"
 
 	"istio.io/istio/tools/istio-iptables/pkg/config"
 	"istio.io/istio/tools/istio-iptables/pkg/constants"
+	dep "istio.io/istio/tools/istio-iptables/pkg/dependencies"
+	"istio.io/pkg/env"
 
 	"github.com/spf13/cobra"
 	"github.com/spf13/viper"
 
 	"istio.io/pkg/log"
 )
 
+var (
+	envoyUserVar = env.RegisterStringVar(constants.EnvoyUser, "istio-proxy", "Envoy proxy username")
```

Comment:
any way we can move this to `constructConfig`. Would like to keep reading config at a single place and also use `viper` as a standard way to read env vars

Reply:
Unfortunately, no. This API is meant to be used this way. See all other usages.
This function must be called at package init time, but at the same time there's no way to retrieve the variable after calling `Register*Var()`. So the value returned by the call must be stored in a variable somewhere. A global variable is the only way to use this API properly.

At least, now those variables will be documented in the HTML docs.